### ISSUE TITLE
[BUG] Running `rancher clusters add-node` in the CLI does not add a new node to an existing Rancher cluster

### ISSUE THREAD

**Issue by markusewalker** [rancher/rancher#38515]
**Rancher Server Setup**
- Rancher version: `v2.6.7-rc1`
- Installation option (Docker install/Helm Chart): N/A
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): N/A
- Proxy/Cert Details: N/A

**Information about the Cluster**
- Kubernetes version: N/A
- Cluster Type (Local/Downstream): N/A
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider)

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom) Admin
  - If custom, define the set of permissions:

**Describe the bug**

After provisioning a cluster in Rancher, when trying to add a new node to that existing Rancher cluster, you're unable to do so using the CLI's `rancher clusters add-node` command. Here is the usage for the command:

```
$ rancher clusters add-node -h
Outputs the docker command needed to add a node to an existing Rancher cluster

Usage: 
  rancher clusters add-node [OPTIONS] [CLUSTERID CLUSTERNAME]

Options:
   --label value   Label to apply to a node in the format [name]=[value]
   --etcd          Use node for etcd
   --management    Use node for management (DEPRECATED, use controlplane instead)
   --controlplane  Use node for controlplane
   --worker        Use node as a worker
   --quiet, -q     Only display IDs or suppress help text
```
When attempting to add, for example, a new node, I am seeing the following behavior:

```
$ rancher clusters add-node --worker mew-rke1
FATA[0000] a node can't be added to the cluster this way 
$ rancher clusters add-node --controlplane mew-rke1
FATA[0000] a node can't be added to the cluster this way 
$ rancher clusters add-node --etcd mew-rke1
FATA[0000] a node can't be added to the cluster this way 
$ rancher clusters add-node --etcd --controlplane --worker mew-rke1
FATA[0000] a node can't be added to the cluster this way 
```
Following the usage exactly, I am unable to add a node of any role as seen above. If there is a different way to add a node, then that needs to be explained further in this command. Either way, I should be able to add a node to my cluster via the CLI without hiccup.

**To Reproduce**

1. Setup Rancher and navigate to the UI in a browser.
2. Be sure to grab the admin user's bearer token to login to the CLI later.
3. Provision a downstream RKE1 or RKE2 node driver cluster using your cloud provider of choice.
4. In a client machine, setup the Rancher CLI.
5. Run `rancher login` and login as an administrative user. Be sure to login to the provisioned cluster you made.
6. Run `rancher clusters add-node --<role> <cluster ID or cluster name>`; you will see the error message.

**Result**
I am unable to add a new node to my existing Rancher cluster.

**Expected Result**

I should be able to add a new node of any role to my existing Rancher cluster.

**Comment by HarrisonWAffel** [comment]
From what I can determine this is functioning as designed. The `clusters add-node` command can be properly used to add nodes to a custom cluster, but is specifically written so that nodes cannot be added to a cluster that utilizes node-pools. Currently, there is no way to manually register a node into a cluster that is not `custom` (i.e utilizes node-pools or GKE, EKS, AKS, etc) within the UI, you must use existing or new node-pools and scale them accordingly. This behavior within the CLI tool is intended to match the behavior of the UI. 

However, I can see how the error message reported is misleading, and that the command usage does not reflect this limitation. The change to prevent the addition of nodes into clusters that utilize node-pools is [here](https://github.com/HarrisonWAffel/cli/commit/db1b64966917a1065c397c0eec213e2bf354ac7f), and similarly states that this was done to match UI functionality. I can improve the usage and reported error message to reflect this.

**Comment by markusewalker** [comment]
Thanks @HarrisonWAffel for providing that insight and the limitation not being CLI-exclusive; that makes sense. Yes, I agree that still we need to improve both the usage and error message. From a customer's perspective, I would assume this functionality is broken when in fact it's working as expected; the clarification will definitely help with the confusion.

**Comment by markusewalker** [comment]
While we wait for the associated PR to be merged, here is the QA test plan:

**TEST PLAN**
1. Setup Rancher and navigate to the UI.
2. Create a standard user, login as that user and generate a bearer token.
3. Provision a downstream RKE2 node driver cluster.
4. Validate that the cluster and nodes appear as `Active` in Cluster Management and Cluster Explorer
5. On a separate client machine, download and setup the Rancher CLI.
6. Run `rancher login -t <standard user token> <Rancher Server URL>` and select a valid project.
7. Run command `rancher clusters add-node --<role> <cluster ID or cluster name>`; validate that the error message you receive is clear.

**Comment by HarrisonWAffel** [comment]
This fix is now in cli v2.6.9-rc1, so this can be tested.

**Comment by markusewalker** [comment]
Verified this has been addressed in `v2.6.9-rc1`.

**ENVIRONMENT**
- Rancher install: Helm
- Rancher version: `v2.6-head` https://github.com/rancher/rancher/commit/e2b5939c41732de974e6eb20e86643af5d64ee7f
- Rancher CLI version: `v2.6.9-rc1`

**TEST RESULT**
PASS

**VERIFICATION STEPS**
1. Setup Rancher, created a standard user, logged in as that user and generated a bearer token.
2. Provisioned a downstream RKE2 node driver cluster; validated that the cluster and nodes came up as `Active` in Cluster Management and Cluster Explorer.
3. On a separate client machine, downloaded and setup the Rancher CLI.
4. Ran `rancher login -t <standard user token> <Rancher Server URL>` and selected a valid project.
5. Ran command `rancher clusters add-node -h`; usage now has the following:
```
$ rancher clusters add-node -h
Outputs the docker command needed to add a node to an existing Rancher custom cluster

Usage: 
  rancher clusters add-node [OPTIONS] [CLUSTERID CLUSTERNAME]

Options:
   --label value   Label to apply to a node in the format [name]=[value]
   --etcd          Use node for etcd
   --management    Use node for management (DEPRECATED, use controlplane instead)
   --controlplane  Use node for controlplane
   --worker        Use node as a worker
   --quiet, -q     Only display IDs or suppress help text
```
6. Ran command `rancher clusters add-node --<role> <cluster ID>` and received much more clear message:
```
$ rancher clusters add-node --worker auto-linode-zoaow
FATA[0000] a node can't be manually registered to a cluster utilizing node-pools 
```
